modules/appeals_api/app/middleware/sidekiq_job_duration_middleware.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'benchmark'
+require 'datadog/statsd'
+
+class SidekiqJobDurationMiddleware
+  # Job duration will only be logged for workers from these modules
+  MODULE_NAMES = %w[AppealsApi VAForms VBADocuments].freeze
+
+  def initialize(_options = {})
+    @statsd = Datadog::Statsd.new
+  end
+
+  # rubocop:disable Style/ExplicitBlockArgument
+  def call(worker, _job, _queue)
+    if MODULE_NAMES.any? { |name| worker.class.name.start_with? name }
+      elapsed_time = Benchmark.realtime do
+        yield
+      end
+
+      Rails.logger.info "Sidekiq job #{worker.class.name} in #{elapsed_time.round} seconds"
+
+      module_name = worker.class.name.split('::').first.underscore
+      @statsd.timing("#{module_name}.sidekiq_job.duration", elapsed_time.round)
```

Comment:
How are we to tell which job has which durations, if we're only grabbing the module name? Shouldn't we also grab the actual class that runs the job?

Reply:
Do we know anyone with read access who could peek for us?

Reply:
manual datadog duration logging isn't necessary :)

<img width="622" alt="Screenshot 2023-04-26 at 15 23 21" src="https://user-images.githubusercontent.com/142862/234716104-f549a298-dcc9-47ca-a312-55580e13e785.png">

Reply:
With the fact that we're getting automatic perform duration logging w/ DataDog - do we actually need this separate logging into rails logs?

Reply:
I'd much rather just look at datadog for this information - if we have a pathway to doing that and it's acceptable to wait to get this information until then, I'm happy to close this

Reply:
I think it was added as a stopgap since we don't have DataDog access yet.

Reply:
It also looks like not all of our jobs are being captured by DataDog, so let's keep this in for now until we know that all our jobs are registered.